Library/Homebrew/bintray.rb

```diff
@@ -131,20 +131,15 @@ def package_exists?(repo:, package:)
     end
   end
 
-  def file_published?(repo:, remote_file:)
+  def check_remote_file(repo:, remote_file:)
     url = "https://dl.bintray.com/#{@bintray_org}/#{repo}/#{remote_file}"
-    begin
-      curl "--fail", "--silent", "--head", "--output", "/dev/null", url
-    rescue ErrorDuringExecution => e
-      stderr = e.output
-                .select { |type,| type == :stderr }
-                .map { |_, line| line }
-                .join
-      raise if e.status.exitstatus != 22 && !stderr.include?("404 Not Found")
-
-      false
+    result = curl_output "--fail", "--silent", "--head", url
+    if result.success?
+      [:present, result.stdout.match(/^X-Checksum-Sha2: ([0-9a-f]{64})/i)[1]]
```

Comment:
```suggestion
      { 
        success: true, 
        checksum: result.stdout.match(/^X-Checksum-Sha2: ([0-9a-f]{64})/i)[1]]
      }
```

or similar; I think a hash makes this a little more readable.

Also: can this fail if it's a success but it can't find the checksum header? What about making the presence of a checksum imply success and checking explicitly for the header before returning a (possibly) missing match?

Reply:
Yeah, I haven't seen any cases where the checksum doesn't exist. But, also the Bintray API documentation is so sparse that it's not clear if the situation where the file exists but doesn't return a checksum is possible.

How about this:

file exists and has checksum: return checksum
file exists but doesn't have checksum: return true
file doesn't exist: return false

That way, a truthy return value always means "the file exists", and you can additionally compare the return value against your local hash to look for a mismatch. (If the remote file exists but doesn't have a checksum, you can't tell if a previous upload already succeeded anyway, so this should be fine).

Reply:
> Yeah, I haven't seen any cases where the checksum doesn't exist. But, also the Bintray API documentation is so sparse that it's not clear if the situation where the file exists but doesn't return a checksum is even possible.

Yeh, with external APIs I always like to assume they will break at any point and start returning garbage 😂 

> file exists and has checksum: return checksum
> file exists but doesn't have checksum: return true
> file doesn't exist: return false

Didn't realise there was essentially three cases here. I'd say in that case return a checksum, empty string, nil (or false)?